tests/contrib/asyncio/test_tracer_safety.py

```diff
@@ -50,19 +47,21 @@ def test_trace_coroutine(self):
         assert "coroutine" == traces[0][0].name
         assert "base" == traces[0][0].resource
 
-    @mark_asyncio
-    def test_trace_multiple_calls(self):
-        @asyncio.coroutine
-        def coro():
-            # another traced coroutine
-            with self.tracer.trace("coroutine"):
-                yield from asyncio.sleep(0.01)
 
-        futures = [asyncio.ensure_future(coro()) for x in range(1000)]
-        for future in futures:
-            yield from future
+@pytest.mark.asyncio
+async def test_trace_multiple_calls(tracer):
+    tracer.configure(context_provider=DefaultContextProvider())
 
-        # the trace is wrong but the Context is finished
-        traces = self.pop_traces()
-        assert 1 == len(traces)
-        assert 1000 == len(traces[0])
+    async def coro():
+        # another traced coroutine
+        with tracer.trace("coroutine"):
+            await asyncio.sleep(0.01)
+
+    futures = [asyncio.ensure_future(coro()) for x in range(1000)]
+    for future in futures:
+        await future
+
+    # the trace is wrong but the Context is finished
+    traces = tracer.pop_traces()
+    assert 1000 == len(traces)
+    assert 1 == len(traces[0])
```

Comment:
I had to change the result here.
I'm not sure if the comment is correct or if my change is wrong?
@Kyle-Verhoog can you weigh in?

Reply:
It's actually weirder: there is now 1k traces with Python 3.9, but Python 3.5 still has the previous 1 traces.